Mathlib/RingTheory/Bialgebra/GroupLike.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2025 Yaël Dillies, Michał Mrugała. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Michał Mrugała
+-/
+import Mathlib.RingTheory.Bialgebra.Equiv
+import Mathlib.RingTheory.Flat.Domain
+
+/-!
+# Group-like elements in a bialgebra
+
+This file defines group-like elements in a bialgebra, ie units `a` such that `Δ a = a ⊗ₜ a`.
+
+The motivating example is the group ring `R[G]`. Group-like elements of `R[G]` are exactly the image
+of `G` inside `R[G]`. We note two things: these group-like elements form a group and are linearly
+independent.
+
+In an arbitrary bialgebra, group-like elements always form a group, and are linearly independent if
+the base ring is a domain.
+
+## Main declarations
+
+* `Bialgebra.IsGroupLikeElem`: Predicate for an element in a bialgebra to be group-like.
+* `Bialgebra.GroupLike`: Subgroup of group-like elements.
+* `Bialgebra.linearIndepOn_isGroupLikeElem`: Group-like elements are linearly independent.
+-/
+
+open Coalgebra Function TensorProduct
+
+variable {F R A B : Type*}
+
+namespace Bialgebra
+section Semiring
+variable [CommSemiring R] [Semiring A] [Semiring B] [Bialgebra R A]
+  [Bialgebra R B] {a b : A}
+
+variable (R) in
+/-- A group-like element in a coalgebra is a unit `a` such that `Δ a = a ⊗ₜ a`. -/
+structure IsGroupLikeElem (a : A) where
+  isUnit : IsUnit a
+  comul_eq_tmul_self : comul (R := R) a = a ⊗ₜ a
+
+@[simp] lemma IsGroupLikeElem.ne_zero [Nontrivial A] (ha : IsGroupLikeElem R a) : a ≠ 0 :=
+  ha.isUnit.ne_zero
```

Comment:
What does tagging this as simp do? Does `simp` now try to infer `Bialgebra R` and create a `IsGroupLikeElem R a` subgoal for every goal of the form `a ≠ 0`?

Reply:
Yes

Reply:
Is this not very bad performance wise?

Reply:
Well, only if you import that lemma, and even then I'm not sure, since synthesizing `HopfAlgebra R A` fails quickly when not available

Reply:
Indeed, performance is quite bad. Let's remove it